src/compiler/bundle/ext-transforms-plugin.ts

```diff
@@ -56,6 +80,22 @@ export const extTransformsPlugin = (
           return null;
         }
 
+        /**
+         * add file to watch list if it is outside of the `srcDir` config path
+         */
+        if (config.watch && (id.startsWith('/') || id.startsWith('.')) && !id.startsWith(config.srcDir)) {
+          compilerCtx.addWatchFile(id.split('?')[0]);
+        }
+
+        /**
+         * initiate map for component styles
+         */
+        const scopeId = getScopeId(data.tag || '', data.mode);
```

Comment:
Well pointed out! This new logic is only important if we transform a style file that is connected to a component. I am not entirely sure when `data` is containing a tag or mode. The transformer isn't used at all when using global styles.

To avoid any issues I moved the code into the `if (data.tag) { ...` block and keep the old behavior if `cmpStyles` is undefined, meaning that `data.tag` had no value.